app/controllers/save_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SaveController < ApplicationController
+  def create
+    @post = Post.find(params[:id])
+
+    if @post.saved == true
+      @post.update(saved: false)
+    else
+      @post.update(saved: true)
+    end
+
+    respond_to do |format|
+      format.js {
+        @post = Post.find(params[:id])
+        @is_saved = @post.saved
+        render "save/create"
```

Comment:
There seems to be a few duplicates in here, can we refactor?

Reply:
Resolved it